R/thumbnail.R

```diff
@@ -0,0 +1,233 @@
+#' Get the Content Thumbnail
+#'
+#' `get_thumbnail` saves the content thumbnail to the given path (default: temp file).
+#' `delete_thumbnail` removes the thumbnail (optionally saving to the given path)
+#' `has_thumbnail` returns whether the content has an thumbnail
+#'
+#' @param content A content object
+#' @param path optional. The path to the image on disk
+#'
+#' @rdname get_thumbnail
+#' @family content functions
+#' @export
+get_thumbnail <- function(content, path = NULL) {
+  validate_R6_class(content, "Content")
+  guid <- content$get_content()$guid
+
+  con <- content$get_connect()
+
+  # Connect 2024.09.0 introduced public endpoints for content thumbnails. We
+  # prefer those, falling back to the unversioned endpoints if unavailable.
+  res <- con$GET(
+    url = con$server_url("content", guid, "__thumbnail__"),
+    parser = NULL
+  )
+  if (httr::status_code(res) == 404) {
+    res <- con$GET(
+      unversioned_url("applications", guid, "image"),
+      parser = NULL
+    )
+  }
+
+  if (httr::status_code(res) == 204) {
+    return(NA)
+  }
+
+  # guess file extension
```

Comment:
If you take my suggestion above and make `path` required, we can delete this whole block. Otherwise, we shouldn't have to guess the file extension--we should know. There's probably a finite set of Content-Types that the Connect API will emit. There's also [this map](https://github.com/yihui/mime/blob/main/R/mimemap.R) of content types to extensions we could reach to if we wanted--`httr` imports `mime` so we already have it available.

Reply:
Going to take your suggestion and make `path` required — people can pass in `tempfile()` if needed. I'm moving the old code to the old `get_image()` function so that still behaves as before.

Reply:
The workflow without `NULL` files is janky if you don't know the format of the image you are about to receive.